scripts/circle-ci-artifacts-utils.js

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env node
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+'use strict';
+
+const {exec} = require('shelljs');
+
+const util = require('util');
+const asyncRequest = require('request');
+const request = util.promisify(asyncRequest);
+
+let circleCIHeaders;
+let jobs;
+let baseTemporaryPath;
+
+async function initialize(circleCIToken, baseTempPath, branchName) {
+  console.info('Getting CircleCI infoes');
+  circleCIHeaders = {'Circle-Token': circleCIToken};
+  baseTemporaryPath = baseTempPath;
+  exec(`mkdir -p ${baseTemporaryPath}`);
+  const pipeline = await _getLastCircleCIPipelineID(branchName);
+  const packageAndReleaseWorkflow = await _getPackageAndReleaseWorkflow(
+    pipeline.id,
+  );
+  _throwIfPendingOrUnsuccessfulWorkflow(packageAndReleaseWorkflow);
+  const testsWorkflow = await _getTestsWorkflow(pipeline.id);
+  _throwIfPendingOrUnsuccessfulWorkflow(testsWorkflow);
+  const jobsPromises = [
+    _getCircleCIJobs(packageAndReleaseWorkflow.id),
+    _getCircleCIJobs(testsWorkflow.id),
+  ];
+
+  const jobsResults = await Promise.all(jobsPromises);
+
+  jobs = jobsResults.flatMap(j => j);
+}
+
+function baseTmpPath() {
+  return baseTemporaryPath;
+}
+
+async function _throwIfPendingOrUnsuccessfulWorkflow(workflow) {
+  if (workflow.status !== 'success') {
+    throw new Error(
+      `The ${workflow.name} workflow status is ${workflow.status}. Please, wait for it to be finished before start testing or fix it`,
+    );
+  }
+}
+
+async function _getLastCircleCIPipelineID(branchName) {
+  const options = {
+    method: 'GET',
+    url: 'https://circleci.com/api/v2/project/gh/facebook/react-native/pipeline',
+    qs: {
+      branch: branchName,
+    },
+    headers: circleCIHeaders,
+  };
+
+  const response = await request(options);
+  if (response.error) {
+    throw new Error(error);
+  }
+
+  const lastPipeline = JSON.parse(response.body).items[0];
+  return {id: lastPipeline.id, number: lastPipeline.number};
```

Comment:
headsup: we need to handle the scenario in which the branch doesn't have a circleCI pipeline (ex. it's a local-only branch) - currently it just fails badly:
<img width="919" alt="Screenshot 2023-07-05 at 14 08 26" src="https://github.com/facebook/react-native/assets/16104054/5e8a99f1-5253-45af-b958-74a84332e2bf">

Reply:
I should have handled this case now.